### PR TITLE
tear_down tests when KeyboardInterrupt

### PR DESCRIPTION
**Pull request by ethanhs**
This is a small fix that I encountered. If you start running tests, and KeyboardInterrupt, the tmp-test-dir has leftover data, as the test is not cleaned up. Jukka suggested assuring that the test was cleaned up as a solution.

### CODE REVIEW

File: mypy/myunit/__init__.py [hunk 1]
```diff
@@ -293,6 +293,7 @@ def run_single_test(name: str, test: Any) -> Tuple[bool, bool]:
         test.run()
     except BaseException as e:
         if isinstance(e, KeyboardInterrupt):
+            test.tear_down()
             raise
         exc_type, exc_value, exc_traceback = sys.exc_info()
     test.tear_down()  # FIX: check exceptions
```
Comment (gnprice) [inline mypy/myunit/__init__.py]: How about let's move this line into a `finally` block? Then we only have to say `tear_down` in one place, and it's easier to see that it'll always be called.

**Comment by gnprice** [comment]
LGTM! Will merge once the AppVeyor build passes.

**Comment by ilevkivskyi** [comment]
The tests have passed, so I assume it is OK to merge this now.

**Comment by gnprice** [comment]
Yep -- thanks @ilevkivskyi!

**Comment by gvanrossum** [comment]
Good catch Ethan, thanks!

**Pull request merged**